File: AddressBook/AddressBook/Dtos/PersonAddress.cs
using AddressBook.Entities;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using AddressBook.Common.Paging;

namespace AddressBook.Dtos
{
    public class PersonAddressRes
    {
        public int PersonAddressId { get; set; }
        public string? Line1 { get; set; }
        public string? Line2 { get; set; }
        public string? PostCode { get; set; }

        // Foreign keys
        public int? CityId { get; set; }
        public CityRes? City { get; set; }

        public int? AddressLabelId { get; set; }
        public AddressLabelRes? AddressLabel { get; set; }

        public int? PersonId { get; set; }
        public PersonRes? Person { get; set; }
    }

    public class PersonAddressReqEdit
    {
        [Required]
        public string? Line1 { get; set; }
        public string? Line2 { get; set; }
        public string? PostCode { get; set; }
        public int? CityId { get; set; }
        public int? AddressLabelId { get; set; }
        public int? PersonId { get; set; }
    }

    public class PersonAddressReqSearch : PagedReq
    {
        [Required]
        public int? PersonId { get; set; }
    }
}

File: AddressBook/react-client/src/dtos/PersonAddress.ts
import Common from "../utility/Common";
import { AddressLabelRes } from "./AddressLabel";
import { CityRes } from "./City";
import { PagedReq } from "./PagedReq";
import { PersonRes } from "./Person";

export interface PersonAddressRes {
  personAddressId?: number;
  line1?: string;
  line2?: string;
  postCode?: string;
  cityId?: number;
  city?: CityRes;
  addressLabelId?: number;
  addressLabel?: AddressLabelRes;
  personId?: number;
  person?: PersonRes;
}

export class PersonAddressReqEdit {
  line1?: string = "";
  line2?: string = "";
  postCode?: string = "";
  cityId?: number = 0;
  addressLabelId?: number = 0;
  personId?: number = 0;
}

export class PersonAddressReqSearch extends PagedReq {
  personId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {personId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.personId = personId;
  }
}